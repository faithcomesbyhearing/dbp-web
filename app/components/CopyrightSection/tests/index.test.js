import React from 'react';
import renderer from 'react-test-renderer';
import {
	copyrights,
	invalidCopyrights,
	invalidCopyrights2,
} from '../../../utils/testUtils/copyrightData';
import CopyrightStatement from '..';

jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));

const props = {
	prefix: 'new',
	copyrights,
};

describe('CopyrightStatement component', () => {
	it('should match previous snapshot for new testament', () => {
		const tree = renderer.create(<CopyrightStatement {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match previous snapshot for old testament', () => {
		const tree = renderer
			.create(<CopyrightStatement {...props} prefix={'old'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match previous snapshot for new testament invalid data', () => {
		const tree = renderer
			.create(
				<CopyrightStatement prefix={'new'} copyrights={invalidCopyrights} />,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match previous snapshot for old testament invalid data', () => {
		const tree = renderer
			.create(
				<CopyrightStatement prefix={'old'} copyrights={invalidCopyrights} />,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match previous snapshot for old testament invalid data where there are no messages', () => {
		const tree = renderer
			.create(
				<CopyrightStatement prefix={'old'} copyrights={invalidCopyrights2} />,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match previous snapshot for no copyright data', () => {
		const tree = renderer
			.create(<CopyrightStatement prefix={'old'} copyrights={{}} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
