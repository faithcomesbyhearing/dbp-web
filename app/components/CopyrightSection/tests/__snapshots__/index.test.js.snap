// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyrightStatement /> handles invalid new-testament data 1`] = `
<DocumentFragment>
  <div
    class="nt-copyright"
  >
    <div
      class="cp-section"
    >
      <p>
        <span>
          Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
        <span>
          Audio: ℗ 2009 Hosanna
        </span>
        <br />
      </p>
    </div>
    <div
      class="cp-section"
    >
      <p>
        <span>
          The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CopyrightStatement /> handles invalid old-testament data 1`] = `
<DocumentFragment>
  <div
    class="ot-copyright"
  >
    <div
      class="cp-section"
    >
      <p>
        <span>
          Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
        <span>
          Audio: ℗ 2009 Hosanna
        </span>
        <br />
      </p>
    </div>
    <div
      class="cp-section"
    >
      <p>
        <span>
          The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CopyrightStatement /> handles old-testament with no messages 1`] = `
<DocumentFragment>
  <div
    class="ot-copyright"
  >
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible audio is brought to you by:
        </span>
      </h3>
      <a
        href="https://www.crossway.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Crossway"
          class="image-landscape fcbh-copyright-logo"
          src="https://images.bible.cloud/logo/crossway.svg"
        />
      </a>
      <h3>
        <span>
          This Bible audio is brought to you by:
        </span>
      </h3>
      <a
        class="org-name"
        rel="noopener"
        target="_blank"
      >
        Good News Publishers
      </a>
    </div>
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        href="https://www.crossway.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Crossway"
          class="image-landscape"
          src="https://images.bible.cloud/logo/crossway.svg"
        />
      </a>
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        class="org-name"
        rel="noopener"
        target="_blank"
      >
        Good News Publishers
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CopyrightStatement /> renders new testament correctly 1`] = `
<DocumentFragment>
  <div
    class="nt-copyright"
  >
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible audio is brought to you by:
        </span>
      </h3>
      <a
        href="http://www.fcbh.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Faith Comes By Hearing"
          class="image-landscape fcbh-copyright-logo"
          src="https://bible.cloud/images/logo/FCBH_Color.svg"
        />
      </a>
      <p>
        <span>
          Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
        <span>
          Audio: ℗ 2009 Hosanna
        </span>
        <br />
      </p>
    </div>
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        href="https://www.crossway.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Crossway"
          class="image-landscape"
          src="https://images.bible.cloud/logo/crossway.svg"
        />
      </a>
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        class="org-name"
        rel="noopener"
        target="_blank"
      >
        Good News Publishers
      </a>
      <p>
        <span>
          The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CopyrightStatement /> renders nothing when no copyright data 1`] = `
<DocumentFragment>
  <div
    class="ot-copyright"
  />
</DocumentFragment>
`;

exports[`<CopyrightStatement /> renders old testament correctly 1`] = `
<DocumentFragment>
  <div
    class="ot-copyright"
  >
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible audio is brought to you by:
        </span>
      </h3>
      <a
        href="http://www.fcbh.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Faith Comes By Hearing"
          class="image-landscape fcbh-copyright-logo"
          src="https://bible.cloud/images/logo/FCBH_Color.svg"
        />
      </a>
      <p>
        <span>
          Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
        <span>
          Audio: ℗ 2009 Hosanna
        </span>
        <br />
      </p>
    </div>
    <div
      class="cp-section"
    >
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        href="https://www.crossway.org"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="Copyright owners logo: Crossway"
          class="image-landscape"
          src="https://images.bible.cloud/logo/crossway.svg"
        />
      </a>
      <h3>
        <span>
          This Bible text is brought to you by:
        </span>
      </h3>
      <a
        class="org-name"
        rel="noopener"
        target="_blank"
      >
        Good News Publishers
      </a>
      <p>
        <span>
          The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
        </span>
        <br />
        <span>
          English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
        </span>
        <br />
      </p>
    </div>
  </div>
</DocumentFragment>
`;
